test/cppsuite/test_harness/workload/database_operation.h

```diff
@@ -48,235 +44,22 @@ class database_operation {
-            /*
-             * Sleep the period defined by the op_rate in the configuration. Do this at the start of
-             * the loop as it could be skipped by a subsequent continue call.
-             */
-            tc->sleep();
-
-            /* Choose a random collection to update. */
-            collection &coll = tc->database.get_random_collection();
-
-            /* Look for existing cursors in our cursor cache. */
-            if (cursors.find(coll.id) == cursors.end()) {
-                log_msg(LOG_TRACE,
-                  "Thread {" + std::to_string(tc->id) +
-                    "} Creating cursor for collection: " + coll.name);
-                /* Open a cursor for the chosen collection. */
-                scoped_cursor cursor = tc->session.open_scoped_cursor(coll.name.c_str());
-                cursors.emplace(coll.id, std::move(cursor));
-            }
-
-            /* Start a transaction if possible. */
-            tc->transaction.try_begin(tc->session.get(), "");
-
-            /* Get the cursor associated with the collection. */
-            scoped_cursor &cursor = cursors[coll.id];
-
-            /* Choose a random key to update. */
-            uint64_t key_id =
-              random_generator::instance().generate_integer<uint64_t>(0, coll.get_key_count() - 1);
-            if (!tc->update(cursor, collection_id, tc->key_to_string(key_id)))
-                continue;
-
-            /* Commit the current transaction if we're able to. */
-            tc->transaction.try_commit(tc->session.get(), "");
-        }
-
-        /* Make sure the last operation is committed now the work is finished. */
-        if (tc->transaction.active())
-            tc->transaction.commit(tc->session.get(), "");
-    }
+    virtual void update_operation(thread_context *tc);
 
     private:
-    static void
-    populate_worker(thread_context *tc)
-    {
-        uint64_t collections_per_thread = tc->collection_count / tc->thread_count;
-        for (int64_t i = 0; i < collections_per_thread; ++i) {
-            collection &coll = tc->database.get_collection((tc->id * collections_per_thread) + i);
-            /*
-             * WiredTiger lets you open a cursor on a collection using the same pointer. When a
-             * session is closed, WiredTiger APIs close the cursors too.
-             */
-            scoped_cursor cursor = tc->session.open_scoped_cursor(coll.name.c_str());
-            for (uint64_t i = 0; i < tc->key_count; ++i) {
-                /* Start a txn. */
-                tc->transaction.begin(tc->session.get(), "");
-                if (!tc->insert(cursor, coll.id, i))
-                    testutil_die(-1, "Got a rollback in populate, this is currently not handled.");
-                tc->transaction.commit(tc->session.get(), "");
-            }
-        }
-        log_msg(LOG_TRACE, "Populate: thread {" + std::to_string(tc->id) + "} finished");
-    }
+    static void populate_worker(thread_context *tc);
```

Comment:
Same. This can probably be moved to the `cxx` as a free function.

Reply:
Done.